aea/helpers/search/models.py

```diff
@@ -417,8 +561,8 @@ def check_validity(self):
                 )
             elif self.type == ConstraintTypes.WITHIN:
                 enforce(
-                    isinstance(self.value, (list, tuple)),
-                    f"Expected one of type in (list, tuple), got {self.value}",
+                    isinstance(self.value, tuple),
```

Comment:
`value` can only be tuple. I removed the list type from being allowed. 

When `value` gets serialised into pb, it becomes `repeated int/double/...`. 
When deserialised, `repeated int/double/...` becomes `tuple[int/float/...]` in python. 

So for an original value to be identical to its serialised-deserialised version, it should be a `tuple`.

Reply:
Similarly in the next couple of lines.

Reply:
ok, fine by me